drivers/flash/nrf_qspi_nor.c

```diff
@@ -410,6 +433,69 @@ static int qspi_wait_while_writing(const struct device *dev)
 	return (ret < 0) ? ret : 0;
 }
 
+static int qspi_wrsrX(const struct device *dev, uint8_t sr_val, uint8_t sr_num)
+{
+	int ret = 0;
+	uint8_t opcode = SPI_NOR_CMD_WRSR;
+	uint8_t length = 1;
+	uint8_t sr_array[2] = {0};
+
+	if (sr_num == 1) {
+		// Writing sr1 clears sr2. Need to read/modify/write both.
+		sr_array[0] = sr_val;
+		if (INST_0_QER == JESD216_DW15_QER_S2B1v1) {
+			ret = qspi_rdsr2(dev);
+			if (ret < 0) {
+				LOG_ERR("RDSR for WRSR failed: %d", ret);
+				return ret;
+			}
+			sr_array[1] = ret;
+			length = 2;
+		}
+	} else if (INST_0_QER == JESD216_DW15_QER_S2B1v1
```

Comment:
This code assumes that `sr_num` is 2 at this point. There is no check for invalid values as in `qspi_rdsrX`. It would be nice to keep consistency in this regard.

Reply:
Agreed. Added a check for sr_num validity.